### ISSUE TITLE
Error starting new project on version 0.17.2

### ISSUE THREAD

**Issue by matheusb-comp** [quasarframework/quasar-cli#154]
I'm getting the same error of Issue #153 on version `0.17.2`, in a new project template.
I believe the error comes from Babel, since they decided to [remove the Stage presets](https://github.com/babel/babel/blob/48fe688a10d83486defdda223ac8d8d88057975d/packages/babel-preset-stage-2/README.md) as of version 7.0.0-beta.54, as stated in the `npm install` warnings below.

However, even after running the [babel-upgrade](https://github.com/babel/babel-upgrade) tool on my project, `quasar-framework`, and `quasar-cli`, I still kept receiving the same errors.

### Software version
```
Operating System         	Linux(4.13.0-45-generic) - linux/x64
NodeJs                   	8.11.3

Global packages          	
  NPM                    	6.2.0
  yarn                   	Not installed
  quasar-cli             	0.17.2
  vue-cli                	2.9.6
  cordova                	Not installed

Important local packages 	
  quasar-cli             	0.17.2	(Quasar Framework CLI)
  quasar-framework       	0.17.2	(Build responsive websites, PWAs, hybrid mobile apps and Electron apps, all simultaneously using same codebase)
  quasar-extras          	2.0.4	(Quasar Framework fonts, icons and animations)
  vue                    	2.5.16	(Reactive, component-oriented view layer for modern web interfaces.)
  vue-router             	3.0.1	(Official router for Vue.js 2)
  vuex                   	3.0.1	(state management for Vue.js)
  electron               	Not installed
  electron-packager      	Not installed
  electron-builder       	Not installed
  @babel/core            	7.0.0-beta.54	(Babel compiler core.)
  webpack                	4.16.2	(Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.)
  webpack-dev-server     	3.1.5	(Serves a webpack app. Updates the browser on changes.)
  workbox-webpack-plugin 	3.4.1	(A plugin for your Webpack build process, helping you generate a manifest of local files that workbox-sw should precache.)
  register-service-worker	1.4.1	(Script for registering service worker, with hooks)
```

#### `npm install` warnings:
```
npm WARN deprecated @babel/preset-stage-2@7.0.0-beta.54: 👋 We've deprecated any official stage presets in favor of users explicitly opt-ing into the proposals they want to use versus any perceived convenience. You can also check https://github.com/babel/babel/issues/7770 for more information.
npm WARN deprecated @babel/preset-stage-3@7.0.0-beta.54: 👋 We've deprecated any official stage presets in favor of users explicitly opt-ing into the proposals they want to use versus any perceived convenience. You can also check https://github.com/babel/babel/issues/7770 for more information.
```

### What did you get as the error?
Error when running `quasar dev`:
```
 ERROR  Failed to compile with 1 errors                                                                                                                                                                                               16:39:49

 error  


✖ ｢wdm｣: Hash: 6d8ed3919a6cca605618
Version: webpack 4.16.2
Time: 15753ms
Built at: 2018-07-26 16:39:49
                                         Asset      Size  Chunks                    Chunk Names
               fonts/KFOmCnqEu92Fr1Mu4mxM.woff  19.4 KiB          [emitted]         
       fonts/flUhRq6tzZclQEJ-Vdg-IuiaDsNa.woff  62.5 KiB          [emitted]         
            fonts/KFOkCnqEu92Fr1MmgVxIIzQ.woff  19.5 KiB          [emitted]         
           fonts/KFOlCnqEu92Fr1MmEU9fBBc-.woff  19.5 KiB          [emitted]         
           fonts/KFOlCnqEu92Fr1MmSU5fBBc-.woff  19.4 KiB          [emitted]         
           fonts/KFOlCnqEu92Fr1MmWUlfBBc-.woff  19.4 KiB          [emitted]         
           fonts/KFOlCnqEu92Fr1MmYUtfBBc-.woff  19.5 KiB          [emitted]         
fonts/flUhRq6tzZclQEJ-Vdg-IuiaDsNcIhQ8tQ.woff2    48 KiB          [emitted]         
                      img/quasar-logo-full.svg  12.6 KiB          [emitted]         
                                        app.js  4.32 MiB     app  [emitted]  [big]  app
                                          0.js  47.9 KiB       0  [emitted]         
                                          1.js  29.8 KiB       1  [emitted]         
                                          2.js  43.3 KiB       2  [emitted]         
                                    index.html  1.32 KiB          [emitted]         
Entrypoint app [big] = app.js
[./.quasar/app.js] 608 bytes {app} [built]
[./.quasar/client-entry.js] 773 bytes {app} [built]
[./node_modules/vue/dist/vue.runtime.esm.js] 163 KiB {app} [built]
[./node_modules/webpack-dev-server/client/index.js?http://0.0.0.0:8080] (webpack)-dev-server/client?http://0.0.0.0:8080 6.33 KiB {app} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 2.64 KiB {app} [built]
[./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 746 bytes {app} [built]
[./node_modules/webpack/hot sync ^\.\/log$] (webpack)/hot sync nonrecursive ^\.\/log$ 170 bytes {app} [built]
[./node_modules/webpack/hot/dev-server.js] (webpack)/hot/dev-server.js 1.51 KiB {app} [built]
[./node_modules/webpack/hot/emitter.js] (webpack)/hot/emitter.js 75 bytes {app} [built]
[./node_modules/webpack/hot/log-apply-result.js] (webpack)/hot/log-apply-result.js 1.3 KiB {app} [built]
[0] multi (webpack)-dev-server/client?http://0.0.0.0:8080 (webpack)/hot/dev-server.js ./.quasar/client-entry.js 52 bytes {app} [built]
[./node_modules/webpack/hot/log.js] (webpack)/hot/log.js 1 KiB {app} [built]
[./src/css/app.styl] 1.38 KiB {app} [built]
[./src/plugins/axios.js] 118 bytes {app} [built]
[./src/plugins/i18n.js] 272 bytes {app} [built]
    + 382 hidden modules

WARNING in ./node_modules/quasar-framework/src/plugins/platform.js 166:8-21
"export 'default' (imported as '_objectSpread') was not found in '@babel/runtime/helpers/objectSpread'
 @ ./node_modules/quasar-framework/src/plugins/notify.js
 @ ./.quasar/import-quasar.js
 @ ./.quasar/app.js
 @ ./.quasar/client-entry.js
 @ multi (webpack)-dev-server/client?http://0.0.0.0:8080 (webpack)/hot/dev-server.js ./.quasar/client-entry.js

[... Various warnings like the one above ...]

ERROR in   Error: webpack-internal:///./node_modules/core-js/modules/_is-object.js:5
  module.exports = function (it) {
                 ^
  TypeError: Cannot assign to read only property 'exports' of object '#<Object>'
  
  - _is-object.js:5 Module.eval
    [.]/[core-js]/modules/_is-object.js:5:16
  
  - _is-object.js:8 eval
    [.]/[core-js]/modules/_is-object.js:8:30
  
  - index.template.html:1767 Module../node_modules/core-js/modules/_is-object.js
    /home/matheus/Documents/quasar/new/src/index.template.html:1767:1
  
  - index.template.html:684 __webpack_require__
    /home/matheus/Documents/quasar/new/src/index.template.html:684:30
  
  - index.template.html:61 fn
    /home/matheus/Documents/quasar/new/src/index.template.html:61:20
  
  - _inherit-if-required.js:1 eval
    [.]/[core-js]/modules/_inherit-if-required.js:1:16
  
  - index.template.html:1743 Object../node_modules/core-js/modules/_inherit-if-required.js
    /home/matheus/Documents/quasar/new/src/index.template.html:1743:1
  
  - index.template.html:684 __webpack_require__
    /home/matheus/Documents/quasar/new/src/index.template.html:684:30
  
  - index.template.html:61 fn
    /home/matheus/Documents/quasar/new/src/index.template.html:61:20
  
  - es6.regexp.constructor.js:3 eval
    [.]/[core-js]/modules/es6.regexp.constructor.js:3:25
```

### What steps did you take, to get the error?
- Install version 0.17.2 of  `quasar-cli`
- Select all options except `Support IE` on `quasar init <folder_name>`
- `cd <folder_name> && npm install`
- `quasar dev`

**Comment by rstoenescu** [comment]
Hi,

Thanks for reporting.

1. It doesn't has anything to do with the preset. The warning can be safely ignored for now.
2. Can't reproduce on a linux machine. But trying...

What is the exact linux distribution and version that you are using?

**Comment by uno-de-piera** [comment]
Same error, I'm unable to continue working with Quasar after upgrade to 0.17 version, now no version of Quasar works on my side.

```
➜  quasar17-2 quasar info

Operating System         	Darwin(16.7.0) - darwin/x64
NodeJs                   	9.11.1

Global packages          	
  NPM                    	5.6.0
  yarn                   	1.7.0
  quasar-cli             	0.17.2
  vue-cli                	2.9.6
  cordova                	8.0.0

Important local packages 	
  quasar-cli             	0.17.2	(Quasar Framework CLI)
  quasar-framework       	0.17.2	(Build responsive websites, PWAs, hybrid mobile apps and Electron apps, all simultaneously using same codebase)
  quasar-extras          	2.0.4	(Quasar Framework fonts, icons and animations)
  vue                    	2.5.16	(Reactive, component-oriented view layer for modern web interfaces.)
  vue-router             	3.0.1	(Official router for Vue.js 2)
  vuex                   	3.0.1	(state management for Vue.js)
  electron               	Not installed
  electron-packager      	Not installed
  electron-builder       	Not installed
  @babel/core            	7.0.0-beta.54	(Babel compiler core.)
  webpack                	4.16.2	(Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.)
  webpack-dev-server     	3.1.5	(Serves a webpack app. Updates the browser on changes.)
  workbox-webpack-plugin 	3.4.1	(A plugin for your Webpack build process, helping you generate a manifest of local files that workbox-sw should precache.)
  register-service-worker	1.4.1	(Script for registering service worker, with hooks)

Networking               	
  Host                   	MacBook-Pro-de-Israel.local
  en1                    	192.168.1.39
  vboxnet0               	192.168.56.1
```

**Comment by rstoenescu** [comment]
@uno-de-piera When you say previous versions of Quasar, what are you referring to?

**This problem is under investigation. It's most likely related to some webpack packages not doing their job correctly.**

**Comment by matheusb-comp** [comment]
@rstoenescu: Thanks for the quick reply. Here are my distribution details:
```
# lsb_release -a
No LSB modules are available.
Distributor ID:	LinuxMint
Description:	Linux Mint 18.2 Sonya
Release:	18.2
Codename:	sonya
# uname -a
Linux <username> 4.13.0-45-generic #50~16.04.1-Ubuntu SMP Wed May 30 11:18:27 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

I've also tried removing the [lazy loading](https://quasar-framework.org/guide/app-lazy-loading---code-splitting.html) of `routes.js` and some suggestions from [other threads](https://github.com/charliekassel/vuejs-datepicker/issues/399) about this [error](https://github.com/webpack/webpack/issues/4039).

Unfortunately I don't understand how all these elements work together, and wasn't able to identify the exact problem. Here is the error in case I run `quasar build`:
```
ERROR in ./node_modules/quasar-extras/material-icons/material-icons.css
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
  TypeError: __webpack_require__(...) is not a function
```

**Comment by uno-de-piera** [comment]
I'm referring that the if I try to install quasar-cli 0.16.0 doesn't works, if I try to install quasar-cli 0.16.4 doesn't works etc, currently I stopped my developments with Quasar because I'm unable to 
make it work.

I can uninstall Quasar 0.17.2, clean the yarn cache and reinstall Quasar 0.16.0, but the issue persists with any version.

Thanks for your help.

**Comment by rstoenescu** [comment]
@uno-de-piera Until I figure this out, delete node_modules **AND yarn.lock or package-json.lock** (depending on whether you use yarn or npm), set your quasar-cli dependency to 0.16.4 then yarn/npm install again.

I am sorry for this trouble, but it is an outside problem, not generated by Quasar. We are looking for solutions.

**Comment by uno-de-piera** [comment]
Thanks @rstoenescu, no problem, don't worry 👍, I have this issue reinstalling all global dependencies and generating a new application with quasar-cli.

```
yarn global remove quasar-cli
yarn cache clean
yarn global add quasar-cli@0.17.2
quasar init new-app
cd new-app
quasar dev -m cordova -T android

ERROR in   Error: webpack-internal:///./node_modules/core-js/modules/_is-object.js:5
  module.exports = function (it) {
                 ^
  TypeError: Cannot assign to read only property 'exports' of object '#<Object>'
```
I tried all possible options before open the issue but currently I don't know what more done :(

**Comment by Akaryatrh** [comment]
You should also remove the `.quasar` folder when switching versions of `quasar-cli`.

**Comment by rstoenescu** [comment]
Your project local quasar-cli version does the work. It doesn't matter the global quasar-cli version installed (as long as it's v0.15+). The global quasar-cli installation is there to be able to quasar init a project folder and for convenience of typing "$ quasar ..." instead of adding npm run scripts in your package.json (global quasar-cli detects if in a Quasar project then defers to the locally installed quasar-cli version).

So if you want to go back to v0.16 in your project, make sure your project's package.json quasar-cli points to 0.16.4.

Thank you for your patience while we figure the v0.17 problem.

**Comment by Akaryatrh** [comment]
@uno-de-piera uninstalling `quasar-cli` won't remove the invisible folder `.quasar` from your project.

**Comment by uno-de-piera** [comment]
I know @Akaryatrh , first I remove the project with rm -rf project and then create a new project :)

**Comment by uno-de-piera** [comment]
Sorry, is normal this behavior?

```
λ iparra [~/tutoriales/quasar] → ls
superapp

λ iparra [~/learn/quasar] → quasar -v
0.16.0

λ iparra [~/learn/quasar] → cd superapp

λ iparra [learn/quasar/superapp] → quasar info

Operating System                Darwin(16.7.0) - darwin/x64
NodeJs                          9.11.1
Global packages
  NPM                           5.6.0
  yarn                          1.7.0
  quasar-cli                    0.17.3
  vue-cli                       3.0.0-rc.7
  cordova                       7.1.0

Important local packages
  quasar-cli                    0.17.3  (Quasar Framework CLI)
  quasar-framework              0.17.2  (Build responsive websites, PWAs, hybrid mobile apps and Electron apps, all simultaneously using same codebase)
  quasar-extras                 2.0.4   (Quasar Framework fonts, icons and animations)
  vue                           2.5.16  (Reactive, component-oriented view layer for modern web interfaces.)
  vue-router                    3.0.1   (Official router for Vue.js 2)
  vuex                          3.0.1   (state management for Vue.js)
  electron                      Not installed
  electron-packager             Not installed
  electron-builder              Not installed
  @babel/core                   7.0.0-beta.53   (Babel compiler core.)
  webpack                       4.16.3  (Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to p
reprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.)
  webpack-dev-server            3.1.5   (Serves a webpack app. Updates the browser on changes.)
  workbox-webpack-plugin        3.4.1   (A plugin for your Webpack build process, helping you generate a manifest of local files that workbox-sw should precache.)
  register-service-worker       1.4.1   (Script for registering service worker, with hooks)

Networking
  Host                          iMac-de-Israel.local
  en1                           192.168.1.41
  vboxnet0                      192.168.56.1


λ iparra [learn/quasar/superapp] → quasar -v
0.17.3

λ iparra [learn/quasar/superapp] → cd ..

λ iparra [~/learn/quasar] → quasar -v
0.16.0
```
I have installed globally quasar 0.16.0, but when I build and enter inside of the project, this contains a quasar app with 0.17.x, strange :(

**Comment by uno-de-piera** [comment]
Now testing with 0.16.0

```
Operating System                Darwin(16.7.0) - darwin/x64
NodeJs                          9.11.1

Global packages
  NPM                           5.6.0
  yarn                          1.7.0
  quasar-cli                    0.16.0
  vue-cli                       3.0.0-rc.7
  cordova                       7.1.0

Important local packages
  quasar-cli                    0.16.0  (Quasar Framework CLI)
  quasar-framework              0.16.0  (Build responsive websites, PWAs, hybrid mobile apps and Electron apps, all simultaneously using same codebase)
  quasar-extras                 2.0.0   (Quasar Framework fonts, icons and i18n.)
  vue                           2.5.16  (Reactive, component-oriented view layer for modern web interfaces.)
  vue-router                    3.0.1   (Official router for Vue.js 2)
  vuex                          3.0.1   (state management for Vue.js)
  electron                      Not installed
  electron-packager             Not installed
  electron-builder              Not installed
  @babel/core                   7.0.0-beta.46   (Babel compiler core.)
  webpack                       4.8.3   (Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to p
reprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.)
  webpack-dev-server            3.1.4   (Serves a webpack app. Updates the browser on changes.)
  workbox-webpack-plugin        3.2.0   (A plugin for your Webpack build process, helping you generate a manifest of local files that workbox-sw should precache.)

Networking
  Host                          iMac-de-Israel.local
  en1                           192.168.1.41
  vboxnet0                      192.168.56.1
```
![image](https://user-images.githubusercontent.com/4218670/43337401-140bb79c-91d4-11e8-899e-898517d45d80.png)

I understand that the problem is the vue-router, but I don't understand why appears now this problem :(

**Comment by matheusb-comp** [comment]
Version 0.17.3 works at a different machine, with the same linux distro and kernel, but with a newer version of NodeJS (9.11.2 instead of 8.11.3)

```
# quasar info
Operating System         	Linux(4.13.0-45-generic) - linux/x64
NodeJs                   	9.11.2

Global packages          	
  NPM                    	6.2.0
  yarn                   	Not installed
  quasar-cli             	0.17.3
  vue-cli                	2.9.6
  cordova                	Not installed

Important local packages 	
  quasar-cli             	0.17.3	(Quasar Framework CLI)
  quasar-framework       	0.17.2	(Build responsive websites, PWAs, hybrid mobile apps and Electron apps, all simultaneously using same codebase)
  quasar-extras          	2.0.4	(Quasar Framework fonts, icons and animations)
  vue                    	2.5.16	(Reactive, component-oriented view layer for modern web interfaces.)
  vue-router             	3.0.1	(Official router for Vue.js 2)
  vuex                   	3.0.1	(state management for Vue.js)
  electron               	Not installed
  electron-packager      	Not installed
  electron-builder       	Not installed
  @babel/core            	7.0.0-beta.53	(Babel compiler core.)
  webpack                	4.16.3	(Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.)
  webpack-dev-server     	3.1.5	(Serves a webpack app. Updates the browser on changes.)
  workbox-webpack-plugin 	3.4.1	(A plugin for your Webpack build process, helping you generate a manifest of local files that workbox-sw should precache.)
  register-service-worker	1.4.1	(Script for registering service worker, with hooks)

# quasar dev
 DONE  Compiled successfully in 9301ms                                                                                                                          21:33:49


 N  App URL........... http://localhost:8080/
    Dev mode.......... spa
    Quasar theme...... mat
    Quasar CLI........ v0.17.3
    Quasar Framework.. v0.17.2
```

**Comment by rstoenescu** [comment]
@matheusb-comp Don't think the Nodejs version matters, as long as it's >= 8. My linux machines all work (different Nodejs versions on them) and can't reproduce the issue. The good news is that tomorrow @pdanpdan will share his linux machine where this problem can be reproduced.

However, can you try with the same Nodejs 8.11.3 on the machine you said it worked? Just to be 110% certain it's not affected by Nodejs version. It shouldn't, but who knows...

**Comment by matheusb-comp** [comment]
Working fine with NodeJS 8.11.3. Thanks for investigating this issue.

**Comment by rstoenescu** [comment]
So it's not the Nodejs version. So what do some linux machines have that others don't, generating this webpack + babel issue? Will try to find out tomorrow, but any help is appreciated.

**Comment by rstoenescu** [comment]
Can you guys try setting quasar-cli dep to "0.17.0-beta.33" and report if it still happens? Thanks.

**Comment by adam-baliatico** [comment]
@rstoenescu no such issues with 0.17.0-beta.33

**Comment by rstoenescu** [comment]
Released CLI v0.17.5 which has the Webpack workaround. Thanks for helping out!

Seems like Webpack needs to explictly be told to exclude node_modules from
Babel transpiling for certain Linux distributions.

**Comment by uno-de-piera** [comment]
Thanks for your help @rstoenescu but not working in my side, the webpack issue has been solved, but now I have a blank screen without errors... it's only happens executing the app with cordova, attached screenshoot.

```
quasar dev -m cordova -T android
```

<img width="2559" alt="quasar-mobile-blank-screen" src="https://user-images.githubusercontent.com/4218670/43364613-cfe2b430-931d-11e8-8c12-0f3cdf0aef89.png">

If execute the app on SPA mode
```
quasar dev
```
Works fine, attached screenshoot

<img width="2560" alt="quasar-spa-mode" src="https://user-images.githubusercontent.com/4218670/43364637-7527f572-931e-11e8-9104-cb31b5d393d0.png">

Also, in the new version (0.17.5), if I selected framework: 'all' option on quasar.config.js the app crashes with the next error.

<img width="1756" alt="quasar-framework-all-crash" src="https://user-images.githubusercontent.com/4218670/43364655-c4000932-931e-11e8-94ce-e3189b5d5dee.png">

My quasar info outputs.

```
Operating System                Darwin(16.7.0) - darwin/x64
NodeJs                          9.11.1

Global packages                 
  NPM                           5.6.0
  yarn                          1.9.2
  quasar-cli                    0.17.5
  vue-cli                       3.0.0-rc.7
  cordova                       7.1.0

Important local packages        
  quasar-cli                    0.17.5  (Quasar Framework CLI)
  quasar-framework              0.17.4  (Build responsive websites, PWAs, hybrid mobile apps and Electron apps, all simultaneously using same codebase)
  quasar-extras                 2.0.4   (Quasar Framework fonts, icons and animations)
  vue                           2.5.16  (Reactive, component-oriented view layer for modern web interfaces.)
  vue-router                    3.0.1   (Official router for Vue.js 2)
  vuex                          3.0.1   (state management for Vue.js)
  electron                      Not installed
  electron-packager             Not installed
  electron-builder              Not installed
  @babel/core                   7.0.0-beta.53   (Babel compiler core.)
  webpack                       4.16.3  (Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.)
  webpack-dev-server            3.1.5   (Serves a webpack app. Updates the browser on changes.)
  workbox-webpack-plugin        3.4.1   (A plugin for your Webpack build process, helping you generate a manifest of local files that workbox-sw should precache.)
  register-service-worker       1.4.1   (Script for registering service worker, with hooks)
```
Thanks!
